### DISCUSSION TITLE
How to declare line break in text message?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [tulir/whatsmeow#417] [WhatsApp protocol Q&A]
I have tried %0a as mentioned in stackover flow for eg: Hello %0a World but its not working out...

**Comment by hrizal** [comment]
"\n"

Pada tanggal Jum, 7 Jul 2023 pukul 01.43 Dineshkumar <
***@***.***> menulis:

**Reply by iamdineshkumar** [reply]
Tried that as well ☹️ %0a %0A /n are the things tried so far but recieved as it's...

![Screenshot_2023-07-07-00-35-25-70.jpg](https://github.com/tulir/whatsmeow/assets/14805534/f40c406f-7379-4843-b0ad-e1cfe931ad20)

**Reply by mamur-rezeki** [reply]
Maybe you need to log the text to see the exact result before sending it.

Common problem is that "\\n" isn't correctly as "\n".

**Comment by hoehermann** [comment]
![image](https://github.com/tulir/whatsmeow/assets/838331/834eb36a-78e0-49d9-8327-a9085c43863d)

Can you show your code? Maybe you are escaping it wrong.

**Reply by mbmisael** [reply]
hello @hoehermann , what did you add as "simple newline" ? Can you please show your code?

**Reply by hoehermann** [reply]
> what did you add as "simple newline"

I press the "enter" key on my keyboard. That adds a new line to the text. I am sorry, but I fail to see the challenge. Here is a simplified example:

	recipient, _ := parseJID(…)
	msg := &waProto.Message{Conversation: proto.String(`A simple newline
	totally works.`)}
	client.SendMessage(context.Background(), recipient, msg)

> Can you please show your code?

The full code is part of my project [purple-gowhatsapp](https://github.com/hoehermann/purple-gowhatsapp). It

1. [takes the user-defined message verbatim](https://github.com/hoehermann/purple-gowhatsapp/blob/1530736af4ac5b93205e182ebd88dc4d86cd1597/src/c/send_message.c#L11),
2. [forwards into the go runtime](https://github.com/hoehermann/purple-gowhatsapp/blob/1530736af4ac5b93205e182ebd88dc4d86cd1597/src/go/bridge.go#L84),
3. [wraps it in a protocol structure](https://github.com/hoehermann/purple-gowhatsapp/blob/1530736af4ac5b93205e182ebd88dc4d86cd1597/src/go/send_message.go#L52),
4. [calls SendMessage](https://github.com/hoehermann/purple-gowhatsapp/blob/1530736af4ac5b93205e182ebd88dc4d86cd1597/src/go/send_message.go#L65).
